### PR TITLE
feat(plugin-auth-backend): resolve CVE-2021-39171

### PR DESCRIPTION
**Pull request by ellisio**
## Hey, I just made a Pull Request!

`passport-saml@^2.0.0` installs a version that flags code scans with [CVE-2021-39171](https://github.com/advisories/GHSA-5379-r78w-42h2). This bumps the package to `3.1.0`, which includes a resolution for this CVE.

Due to the `2.0.0` -> `3.1.2` upgrade, the `auth.saml.cert` config parameter is now required.

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by ellisio** [comment]
Looking into the typing issue now.

File: plugins/auth-backend/src/providers/saml/provider.ts [hunk 1]
```diff
@@ -119,14 +119,14 @@ export type SamlProviderOptions = {};
...
+  return ({ providerId, globalConfig, config, tokenIssuer }) => {
     const opts = {
       callbackUrl: `${globalConfig.baseUrl}/${providerId}/handler/frame`,
       entryPoint: config.getString('entryPoint'),
       logoutUrl: config.getOptionalString('logoutUrl'),
       issuer: config.getString('issuer'),
-      cert: config.getOptionalString('cert'),
-      privateCert: config.getOptionalString('privateKey'),
+      cert: config.getString('cert'),
+      privateKey: config.getOptionalString('privateKey'),
```
Comment (ellisio) [inline plugins/auth-backend/src/providers/saml/provider.ts:129]: `privateCert` was deprecated, and removed, [here](https://github.com/node-saml/passport-saml/pull/591).

**Comment by freben** [comment]
Hi! Thanks for looking at this. Haven't reviewed yet, but see https://github.com/backstage/backstage/pull/7015 which addresses the same. Due to the stability level of this package and the breaking change that this implies, it's awaiting merge a little longer.

**Comment by benjdlambert** [comment]
@freben I'm wondering if because it's a CVE do we not just merge the fix and get it out sooner rather than later and just accept that we broke the stability index because of a security issue? Not really sure what other projects do here tbf.

**Comment by ellisio** [comment]
@freben your change is identical. I don't know why I couldn't find it before I did this work ha. Yours has a better change log message too, so I'll close this out in favor of yours.

🍻

**Pull request closed without merging**